app/controllers/integrations/base_controller.rb

```diff
@@ -111,17 +111,20 @@ def service_name
     @service_name ||= self.class.name.demodulize.sub('Controller', '').downcase
   end
 
-  def create_docker_image
-    options = {git_sha: commit}
+  def create_docker_images
     scope = project.builds
-    return if scope.where(options).first
-    build = project.builds.create!(options.merge(
-      git_ref: branch,
-      description: message,
-      creator: user,
-      label: branch
-    ))
-    DockerBuilderService.new(build).run(push: true, tag_as_latest: true)
+    project.dockerfile_list.each do |dockerfile|
+      options = {git_sha: commit, dockerfile: dockerfile}
+      next if scope.where(options).first
+
+      build = project.builds.create!(options.merge(
+        git_ref: branch,
+        description: message,
+        creator: user,
+        label: branch
+      ))
+      DockerBuilderService.new(build).run(push: true, tag_as_latest: true)
```

Comment:
Remind me, does calling `run` block, or return almost immediately?  We might want to create all the builds first, and then build them in a separate loop.

Reply:
it backgrounds the actual work ... so this should work just fine